ws peer dependency fix

rBangay opened the pull request:
## What does this change?
security fix: force `ws` peer dependency to use a patched/fixed version

tjmw commented on yarn.lock:
```diff
@@ -16592,27 +17125,10 @@ write-file-atomic@^4.0.2:
-  resolved "https://registry.yarnpkg.com/ws/-/ws-8.13.0.tgz#9a9fb92f93cf41512a0735c8f4dd09b8a1211cd0"
-  integrity sha512-x9vcZYTrFPC7aSIbj7sRCYo7L/Xb8Iy+pW0ng0wt2vCJv7M9HOMy0UoN3rr+IFC7hb7vXoqS+P9ktyLLLhO+LA==
+ws@>=7.5.10, ws@>=8.17.1, ws@^6.1.0, ws@^7.3.1, ws@^8.11.0, ws@^8.13.0, ws@^8.2.3:
+  version "8.17.1"
```
Maybe I shouldn't be worrying too much about yarn internals but I'm confused about how `8.17.1` satisfies all the versions listed above. E.g. `ws@^6.1.0` doesn't seem like it should be satisfied with `8.17.1`.

rBangay replied:
It's a good point, I may have been too greedy with the globbing in the `resolutions` block in package.json .. I'll see if I can be more specific where multiple peer dependency versions are used

tjmw replied:
Ah right sorry I hadn't quite appreciated that change in the `resolutions` block for some reason! Yeah could be worth being more specific (it looks like both v7 and v8, which are affected, have a patch available). So it might be just a case of making sure any v7 or v8 in the project are on the latest version of that major release. Having said all that, if it works then 👍

rBangay commented:
> Could be worth co-ordinating with @michaelbjacobson as he's working on a major storybook upgrade.

I think I've fixed the specific major versions the peer dependencies should use. It looks like you can be way more specific in the resolutions block, which hopefully will make Michaels life a lot easier

rBangay commented:
> > Could be worth co-ordinating with @michaelbjacobson as he's working on a major storybook upgrade.
> 
> I think I've fixed the specific major versions the peer dependencies should use. It looks like you can be way more specific in the resolutions block, which hopefully will make Michaels life a lot easier

Michaels pr went out first in the end, so I've rebased this branch and made a slight change (webpack-dev-server had been updated and no longer needs the version of ws it uses to be patched)
